prov/shm/src/smr_init.c

```diff
@@ -72,6 +72,48 @@ static void smr_resolve_addr(const char *node, const char *service,
 	(*addr)[*addrlen - 1]  = '\0';
 }
 
+static uint64_t smr_calculate_shm_usage(size_t tx_count, size_t rx_count)
+{
+	size_t total_size;
+	size_t tx_size, rx_size;
+
+	tx_size = roundup_power_of_two(tx_count);
+	rx_size = roundup_power_of_two(rx_count);
+
+	total_size = sizeof(struct smr_region) +
+		     sizeof(struct smr_cmd_queue) + sizeof(struct smr_cmd) * rx_size +
+		     sizeof(struct smr_resp_queue) + sizeof(struct smr_resp) * tx_size +
+		     sizeof(struct smr_inject_pool) + sizeof(struct smr_inject_pool_entry) * rx_size +
+		     sizeof(struct smr_sar_pool) + sizeof(struct smr_sar_pool_entry) * SMR_MAX_PEERS +
```

Comment:
Ideally, this would have some limit, such as MAX_SAR_PER_PEER x MAX_PEERS

Reply:
Sorry, I'm confused here. total_size calculated in smr_calculate_shm_usage (will be rename) is the shm size we need per peer. 
It's the same value as the one in smr_create(), which is for per peer.